lib/sensu-handler.rb

```diff
@@ -79,17 +79,25 @@ def bail(msg)
       exit 0
     end
 
+    # Return a hash of API settings derived first from ENV['SENSU_API_URL'] if set,
+    # then Sensu config `api` scope if configured, and finally falling back to
+    # to ipv4 localhost address on default API port.
+    #
+    # @return [Hash]
     def api_settings
-      @api_settings ||= if ENV['SENSU_API_URL']
+      case
+      when ENV['SENSU_API_URL']
         uri = URI(ENV['SENSU_API_URL'])
-        {
+        @api_settings = {
           'host' => uri.host,
           'port' => uri.port,
           'user' => uri.user,
           'password' => uri.password
         }
       else
-        settings['api']
+        @api_settings = settings['api'] || {}
+        @api_settings['host'] ||= '127.0.0.1'
+        @api_settings['port'] ||= 4567
       end
     end
```

Comment:
Perhaps we should add `@api_settings` on the last line of the method to imply that it returns its value? If `api_settings()` is the accessor, the first line of the method should be `return @api_settings if @api_settings`.